src/Build/BackEnd/Components/Caching/ResultsCache.cs

```diff
@@ -163,15 +163,15 @@ public ResultsCacheResponse SatisfyRequest(BuildRequest request, List<string> co
                 if (_resultsByConfiguration.TryGetValue(request.ConfigurationId, out BuildResult allResults))
                 {
                     // Check for targets explicitly specified.
-                    bool explicitTargetsSatisfied = CheckResults(allResults, request.Targets, response.ExplicitTargetsToBuild, skippedResultsDoNotCauseCacheMiss);
+                    bool explicitTargetsSatisfied = CheckResults(request.BuildRequestDataFlags, allResults, request.Targets, response.ExplicitTargetsToBuild, skippedResultsDoNotCauseCacheMiss);
```

Comment:
This check (along with the other two you changed) are specific to targets; I don't think it makes sense to check BuildRequestDataFlags here as well. I'd have a separate check either above or below this if block for that.

Reply:
@ladipro , I put it in CheckResults method because I wanted to report it as a part of [targetsMissingResults](https://github.com/dotnet/msbuild/blob/995ca63dbd7104208e33a6ff8dffacab5daa44d2/src/Build/BackEnd/Components/Caching/ResultsCache.cs#L312)
Do I need to track it in my separate method?

Reply:
[targetsMissingResults](https://github.com/dotnet/msbuild/blob/995ca63dbd7104208e33a6ff8dffacab5daa44d2/src/Build/BackEnd/Components/Caching/ResultsCache.cs#L312) is unused and should be possible to eliminate together with `ExplicitTargetsToBuild`.

I think it makes sense to check the flags first and only if they match proceed to checking targets, just as a tiny perf optimization.